src/main/java/no/sikt/sws/OpenSearchClient.java

```diff
@@ -84,11 +76,19 @@ public boolean needsConnectionLeftOpen() {
         }
     };
 
-    public OpenSearchResponse sendRequest(HttpMethodName httpMethod, String path, String data) throws IOException {
+    public OpenSearchResponse sendRequest(HttpMethodName httpMethod, String path, String data) {
 
         Request<Void> request = new DefaultRequest<>("es");
         request.setHttpMethod(httpMethod);
-        request.setEndpoint(buildUri(path));
+
+        logger.info(path);
+        var uri = buildUri(path);
+
+        request.setEndpoint(uri);
+
+        if (uri.getQuery() != null) {
+            request.setParameters(getStringListMap(uri));
+        }
```

Comment:
I suspect that with this approach we might be adding the parameters twice. 
You are still doing the request.setEndpoint(buildUri(path)) although you refactored it, but you are also doing request.setParameters() afterwards.
Perhaps we need to to something like
`request.setEndpoint(uri.getPath())`